backend/routes.js

```diff
@@ -594,40 +637,86 @@ route.POST('/deleteFile/{hash}', {
 
   // Authentication passed, now proceed to delete the file and its associated
   // keys
-  const rawManifest = await sbp('chelonia.db/get', hash)
-  if (!rawManifest) return Boom.notFound()
   try {
-    const manifest = JSON.parse(rawManifest)
-    if (!manifest || typeof manifest !== 'object') return Boom.badData('manifest format is invalid')
-    if (manifest.version !== '1.0.0') return Boom.badData('unsupported manifest version')
-    if (!Array.isArray(manifest.chunks) || !manifest.chunks.length) return Boom.badData('missing chunks')
-    // Delete all chunks
-    await Promise.all(manifest.chunks.map(([, cid]) => sbp('chelonia.db/delete', cid)))
+    await sbp('backend/deleteFile', hash)
+    return h.response()
   } catch (e) {
-    console.warn(e, `Error parsing manifest for ${hash}. It's probably not a file manifest.`)
-    return Boom.notFound()
+    return errorMapper(e)
   }
-  // The keys to be deleted are not read from or updated, so they can be deleted
-  // without using a queue
-  await sbp('chelonia.db/delete', hash)
-  await sbp('chelonia.db/delete', `_private_owner_${hash}`)
-  await sbp('chelonia.db/delete', `_private_size_${hash}`)
-  await sbp('chelonia.db/delete', `_private_deletionToken_${hash}`)
-  const resourcesKey = `_private_resources_${owner}`
-  // Use a queue for atomicity
-  await sbp('okTurtles.eventQueue/queueEvent', resourcesKey, async () => {
-    const existingResources = await sbp('chelonia.db/get', resourcesKey)
-    if (!existingResources) return
-    if (existingResources.endsWith(hash)) {
-      await sbp('chelonia.db/set', resourcesKey, existingResources.slice(0, -hash.length - 1))
-      return
+})
+
+route.POST('/deleteContract/{hash}', {
+  auth: {
+    // Allow file deletion, and allow either the bearer of the deletion token or
+    // the file owner to delete it
+    strategies: ['chel-shelter', 'chel-bearer'],
+    mode: 'required'
+  }
+}, async function (request, h) {
```

Comment:
Question: as this is a long-running process, should we immediately mark this contract as "disabled" or "don't touch" or whatever so that, for example, if someone tries to do a lookup on someone's identity contract (e.g. grabs their username and then fetches their contract and starts loading it) while this contract is being deleted, they are instead immediately presented with 410? Perhaps we should immediately mark the name=>CID lookup as dead?

Or do you think it's fine to leave as-is?

Reply:
I think this is fine to leave as is. The contract will effectively become read-only soon (as it will be deleted after its descendants are scheduled to be deleted too) and then eventually deleted.